common/src/main/java/org/apache/celeborn/common/network/client/TransportClient.java

```diff
@@ -225,14 +226,25 @@ public ChannelFuture pushData(
     PushRequestInfo info = new PushRequestInfo(dueTime, callback);
     handler.addPushRequest(requestId, info);
     pushData.requestId = requestId;
-    PushChannelListener listener = new PushChannelListener(requestId, rpcSendoutCallback);
+    PushChannelListener listener =
+        new PushChannelListener(requestId, rpcSendoutCallback, rpcFailureCallback);
     ChannelFuture channelFuture = channel.writeAndFlush(pushData).addListener(listener);
     info.setChannelFuture(channelFuture);
     return channelFuture;
   }
 
   public ChannelFuture pushMergedData(
       PushMergedData pushMergedData, long pushDataTimeout, RpcResponseCallback callback) {
+    return pushMergedData(
+        pushMergedData, pushDataTimeout, callback, null, () -> pushMergedData.body().release());
```

Comment:
I think this could introduce IllegalReferenceCountException. Maybe here needs more tests

Reply:
How about current?